File: Js/FirstLast.js
var searchRange = function (nums, target) {
  let start = 0
  let end = nums.length - 1
  let item = [-1, -1]

  if (start == end) {
    if (nums[0] == target) {
      return [0, 0]
    } else {
      return [-1, -1]
    }
  }
  while (start <= end) {
    mid = parseInt(start + (end - start) / 2)

    if (target < nums[mid]) {
      end = mid - 1
    } else if (target > nums[mid]) {
      start = mid + 1
    } else {
      break
    }
  }
  if (start > end) {
    return [-1, -1]
  }

  for (let index = mid; index >= start; index--) {
    if (nums[index] < target) {
      item[0] = index + 1
      break
    } else if (nums[index] == target) {
      item[0] = index
    }
  }

  for (let index = mid; index <= end; index++) {
    if (nums[index] > target) {
      item[1] = index - 1
      break
    } else if (nums[index] == target) {
      item[1] = index
    } else {
      if (index > 0) {
        item = index - 1
      }
    }
  }
  return item
}
nums = [2, 2]
target = 2
let res = searchRange(nums, target)
console.log(res)

File: python/FirstLast.py
class Solution():
    def searchRange(self, nums, target):
        start=0
        end=len(nums)-1
        item=[-1,-1]

        if(start==end):
            if(nums[0]==target):
                return [0,0]
            else:
                return item

        while(start <=end):
            mid=start+(end-start)//2
            if(target < nums[mid]):
                end=mid-1
            elif(target>nums[mid]):
                start=mid+1
            else:
                break
        
        if(start >end):
            return item

        for i in range(mid,start-1,-1):
            if(nums[i]<target):
                item[0]=i+1
                break
            elif(nums[i]==target):
                item[0]=i
        for i in range(mid,end+1,1):
            if(nums[i] >target):
                item[1]=i-1
                break
            elif(nums[i]==target):
                item[1]=i
            else:
                if(i>0):
                    item[1]=i-1
        print(item)
        return item


nums = [5,7,7,8,8,10]
target=6
tx=Solution()
tx.searchRange(nums,target)
